ee/modules/600-flant-integration/images/flant-pricing/hooks/controller_metrics.py

```diff
@@ -0,0 +1,189 @@
+#!/usr/bin/env python3
+#
+# Copyright 2023 Flant JSC
+# Licensed under the Deckhouse Platform Enterprise Edition (EE) license.
+# See https://github.com/deckhouse/deckhouse/blob/main/ee/LICENSE
+#
+# This hook is responsible for generating metrics for d8 controllers resource consumption.
+
+
+from dataclasses import dataclass
+from abc import ABC, abstractmethod
+from typing import List, Dict, Any, TypeVar
+
+from shell_operator import hook
+from utils import prometheus_query_value, prometheus_metric_builder, prometheus_function_builder
+
+
+@dataclass
+class Controller:
+    name: str
+    namespace: str
+    kind: str
+    module: str
+
+
+MetricCollectorT = TypeVar("MetricCollectorT", bound="AbstractMetricCollector")
+
+
+class AbstractMetricCollector(ABC):
+
+    metric_group = "group_d8_controller_metrics"
+    cpu_metric_name = "flant_pricing_controller_average_cpu_usage_seconds"
+    memory_metric_name = "flant_pricing_controller_average_memory_working_set_bytes:without_kmem"
+    interval = "5m"
+
+    def collect(self, ctx: hook.Context, controllers: List[Controller]):
+        '''Export metrics to hook context from Controllers list'''
+
+        ctx.metrics.expire(self.metric_group)
```

Comment:
Labels `"labels": controller_metric_labels(ctrl)` are the same. I'd write them here once, and would not extract a function. We can reuse variable instead of a function, so all the metric fields would be in one place.